x-pack/plugins/watcher/server/plugin.ts

```diff
@@ -45,21 +46,26 @@ export class WatcherServerPlugin implements Plugin<void, void, any, any> {
     this.log = ctx.logger.get();
   }
 
-  async setup(
-    { http, elasticsearch: elasticsearchService }: CoreSetup,
-    { licensing }: Dependencies
-  ) {
+  async setup({ http, getStartServices }: CoreSetup, { licensing }: Dependencies) {
     const router = http.createRouter();
     const routeDependencies: RouteDependencies = {
       router,
       getLicenseStatus: () => this.licenseStatus,
     };
 
-    const config = { plugins: [elasticsearchJsPlugin] };
-    const watcherESClient = elasticsearchService.createClient('watcher', config);
+    const getWatcherEsClient = once(async () => {
```

Comment:
are we still okay to provide `registerRouteHandlerContext` during the `start` phase? that would simplify this code

Reply:
`registerRouteHandlerContext` from start could potentially introduce a race condition where HTTP route handlers are being called before the call to `registerRouteHandlerContext` finished registering the context.